controllers/storagecluster/storageconsumer.go

```diff
@@ -130,3 +108,132 @@ func (s *storageConsumer) ensureDeleted(_ *StorageClusterReconciler, _ *ocsv1.St
+		storageClassNames[util.GenerateNameForNonResilientCephBlockPoolStorageClass(storageCluster)] = true
+	}
+
+	if storageCluster.Spec.NFS != nil && storageCluster.Spec.NFS.Enable {
+		storageClassNames[util.GenerateNameForCephNetworkFilesystemStorageClass(storageCluster)] = true
+	}
+
+	if storageCluster.Spec.Encryption.StorageClass && storageCluster.Spec.Encryption.KeyManagementService.Enable {
+		storageClassNames[util.GenerateNameForEncryptedCephBlockPoolStorageClass(storageCluster)] = true
+	}
+
+	// for day2 storageclasses
+	storageClassesInCluster := &storagev1.StorageClassList{}
+	if err := kubeClient.List(ctx, storageClassesInCluster); err != nil {
+		return nil, err
+	}
+	for idx := range storageClassesInCluster.Items {
+		// TODO: skip storageclasses that are from external mode if both internal & external mode is enabled
+		sc := &storageClassesInCluster.Items[idx]
+		scName := sc.Name
+		scProvisioner := sc.Provisioner
+		if slices.Contains(supportedCsiDrivers, scProvisioner) && !storageClassNames[scName] {
+			storageClassNames[scName] = true
+		}
+	}
+
+	scSpec := make([]ocsv1a1.StorageClassSpec, len(storageClassNames))
+	idx := 0
+	for scName := range maps.Keys(storageClassNames) {
+		// TODO: store the spec as the value which allows each value to be customizable corresponding to the name,
+		// not doing now as we only have name in the storageclassspec
+		scSpec[idx] = ocsv1a1.StorageClassSpec{Name: scName}
+		idx++
+	}
+	return scSpec, nil
+}
+
+func getVolumeSnapshotClassesForDistribution(ctx context.Context, kubeClient client.Client, storageCluster *ocsv1.StorageCluster) (
+	[]ocsv1a1.VolumeSnapshotClassSpec, error) {
+
+	volumeSnapshotClassNames := map[string]bool{}
+	volumeSnapshotClassNames[util.GenerateNameForSnapshotClass(storageCluster.Name, util.RbdSnapshotter)] = true
+	volumeSnapshotClassNames[util.GenerateNameForSnapshotClass(storageCluster.Name, util.CephfsSnapshotter)] = true
+
+	if storageCluster.Spec.NFS != nil && storageCluster.Spec.NFS.Enable {
+		volumeSnapshotClassNames[util.GenerateNameForCephNetworkFilesystemStorageClass(storageCluster)] = true
+	}
+
+	// for day2 volumesnapshotclasses
+	volumeSnapshotClassesInCluster := &snapapi.VolumeSnapshotClassList{}
+	if err := kubeClient.List(ctx, volumeSnapshotClassesInCluster); err != nil {
+		return nil, err
+	}
+	for idx := range volumeSnapshotClassesInCluster.Items {
+		// TODO: skip volumesnapshotclasses that are from external mode if both internal & external mode is enabled
+		vsc := &volumeSnapshotClassesInCluster.Items[idx]
+		vscName := vsc.Name
+		vscDriver := vsc.Driver
+		if slices.Contains(supportedCsiDrivers, vscDriver) && !volumeSnapshotClassNames[vscName] {
+			volumeSnapshotClassNames[vscName] = true
+		}
+	}
+
+	vscSpec := make([]ocsv1a1.VolumeSnapshotClassSpec, len(volumeSnapshotClassNames))
+	idx := 0
+	for vscName := range maps.Keys(volumeSnapshotClassNames) {
+		vscSpec[idx] = ocsv1a1.VolumeSnapshotClassSpec{Name: vscName}
+		idx++
+	}
```

Comment:
As the wanted capacity is known there is no point in using an index. Using append is more idiomatic and with a known capacity will not re-allocate

```suggestion
	vscSpec := make([]ocsv1a1.VolumeSnapshotClassSpec, 0, len(volumeSnapshotClassNames))
	for vscName := range maps.Keys(volumeSnapshotClassNames) {
		vscSpec = append(csvSpec, ocsv1a1.VolumeSnapshotClassSpec{Name: vscName})
	}
```

Reply:
done.